src/db.cpp

```diff
@@ -5069,28 +5143,76 @@ void load_consist(void)
               auto_repair_obj(obj, buf3);
 
             inside = data.GetInt(buf, 0);
-            if (inside > 0) {
-              if (inside == last_in)
-                last_obj = last_obj->in_obj;
-              else if (inside < last_in)
-                while (inside <= last_in) {
-                  if (!last_obj) {
-                    snprintf(buf2, sizeof(buf2), "Load error: Nested-item save failed for %s. Disgorging to room.", GET_OBJ_NAME(obj));
-                    mudlog(buf2, NULL, LOG_SYSLOG, TRUE);
-                    break;
+            if (house_version == VERSION_HOUSE_FILE) {
```

Comment:
Wouldn't this be the kind of place where you want to hardcode the version rather than put it in a global like this? My concern is that if the version is changed to 2, the maintainer will update VERSION_HOUSE_FILE to reflect that, and then this v1 code will no longer work for it. Might be better to do `if (house_version == VERSION_HOUSE_FILE_V1)` or something.

Reply:
The intent is to document format changes in the header. `VERSION_HOUSE_FILE` should always point to current and with any updates to the format old versions that need to be handled can be added to the header (as eg `VERSION_HOUSE_FILE_V1` and obviously the logic modified as it will need handling the extra changes anyway--which means replacing macros where needed). The intended design would be more obvious if I had a previous version available to list at the header but currently it's just 1 and none (with defaults to 0).

I did not hardcode them on the spot because if changes grow in number it'll be hard to document it as there would be version numbers there won the spot without much explanation (or multiple copies of comments--like I have now so someone can read what's going on without having to guess there's a comment at a different instance of the algo--that multiplied heh).
I plan to document this entirely in the header to keep everything consistent. By reading the comment there someone looking at the code for the first time should be able to figure out how to implement format changes and how to handle old ones. If someone doesn't understand how to handle that part properly objects could go poof.
 I wanted you to look at it though before I start documenting it.

Reply:
I'm not 100% sure I'm understanding this, let me try to rephrase my point and see if it makes sense.

Under imaginary logic v1, you have a field that's an int, and that works fine-- VERSION_HOUSE_FILE is 1, and this is a version-1 file, so it's all kosher.

Once you upgrade to logic v2, which changes that same field to a string, the parsing for v2 won't work on a v1 file. Thus, the coder potentially needs to remember to move all the logic from this current block to a new block with `if (house_version == 1)` (because this is the logic that's known good for version 1), then write the new logic in the VERSION_HOUSE_FILE block that's compatible with v2.

If instead this block was always named `if (house_version == v1)`, then there's no chance of forgetting that, and the coder can write a new `== v2` block above it with their new logic.

Reply:
Yeah this works but there's no central documentation location in code that way. It's meant to look like this:

```
/* Current versions are defined in VERSION_HOUSE_FILE and VERSION_VEH_FILE
 * Making changes to the formats you need to change the current version macro
 * to the new version, which is what is written to file,  add a new macro
 * for the previous version, document the format changes here and handle them 
 * accordingly in code.
 */
// Current version adds X
#define VERSION_HOUSE_FILE 3
// Current version adds Y
#define VERSION_VEH_FILE 2

// Adds Blahblah required field.
#define VERSION_HOUSE_FILEV2
// Saves the objects in reverse with the container last. Then reads objects in turn, saves
// anything with a higher nesting level than 0 in a vector, then once a container is found
// places all objects with nesting level +1 into it.
#define VERSION_HOUSE_FILEV1
// Original logic, saves containers last and reads them first, then places subsequent objects
// with nesting level +1 into them. No version number written in file so this defaults to 0.
#define VERSION_HOUSE_FILEV0

// Saves the objects in reverse with the container last. Then reads objects in turn, saves
// anything with a higher nesting level than 0 in a vector, then once a container is found
// places all objects with nesting level +1 into it.
#define VERSION_VEH_FILEV1
// Original logic, saves containers last and reads them first, then places subsequent objects
// with nesting level +1 into them. No version number written in file so this defaults to 0.
#define VERSION_VEH_FILEV0```

So basically when you read the code and hit upon an macro, you go to its definition and got complete documentation for everything. Current version is what's written to the files, additional versions that need special handling are as macros with documentation below. Then change version handling accordingly using the defined macros as something may be something like an just an if block inside the current handling. Makes sense?

Reply:
Do note that I absolutely loathe using macros in general :P I do hate our getter/setter ones with passion haha. But it is one of those cases where it makes sense. Still, if you don't like this design let me know and I'll change it.